include/RAJA/policy/openmp/forall.hpp

```diff
@@ -55,16 +55,20 @@ namespace omp
 /// OpenMP parallel policy implementation
 ///
 template <typename Iterable, typename Func, typename InnerPolicy>
-RAJA_INLINE void forall_impl(const omp_parallel_exec<InnerPolicy>&,
-                             Iterable&& iter,
-                             Func&& loop_body)
+RAJA_INLINE resources::EventProxy<resources::Host> forall_impl(resources::Host &host_res,
+                                                    const omp_parallel_exec<InnerPolicy>&,
+                                                    Iterable&& iter,
+                                                    Func&& loop_body)
 {
+  resources::EventProxy<resources::Host> ep(&host_res);
 
   RAJA::region<RAJA::omp_parallel_region>([&]() {
     using RAJA::internal::thread_privatize;
     auto body = thread_privatize(loop_body);
-    forall_impl(InnerPolicy{}, iter, body.get_priv());
+    ep = forall_impl(host_res, InnerPolicy{}, iter, body.get_priv());
```

Comment:
Is this a race condition?

Reply:
Yep, I can just return the event at the end of this function anyway

Reply:
Done.